aptos-move/aptos-aggregator/src/types.rs

```diff
@@ -212,7 +214,26 @@ impl TryIntoMoveValue for DelayedFieldValue {
         Ok(match (self, layout) {
             (Aggregator(v) | Snapshot(v), U64) => Value::u64(v as u64),
             (Aggregator(v) | Snapshot(v), U128) => Value::u128(v),
-            (Derived(bytes), layout) if is_string_layout(layout) => bytes_to_string(bytes),
+            (Derived(bytes), layout) if is_string_layout(layout) => {
+                if bytes.len() > STRING_AGG_FIXED_LEN {
+                    return Err(
+                        PartialVMError::new(StatusCode::VM_EXTENSION_ERROR).with_message(format!(
+                            "string snapshot too large: {:?}",
+                            String::from_utf8(bytes)
```

Comment:
nit: maybe we should not print string if it is to large :)

Reply:
Things will crash,  so fine to print and die